### PR TITLE
Model docs & fixes(remake)

### PR DESCRIPTION
**Pull request by Shqrdx**
remake of https://github.com/mono/MonoGame/pull/3738

### CODE REVIEW

**Comment by Shqrdx** [comment]
@tomspilman Accept ?

File: MonoGame.Framework/Graphics/Model.cs [hunk 1]
```diff
@@ -1,44 +1,49 @@
...
+        /// </summary>
+        public object Tag { get; set; }
+
+		internal Model()
 		{
 
 		}
 
-		public Model(GraphicsDevice graphicsDevice, List<ModelBone> bones, List<ModelMesh> meshes)
+        internal Model(GraphicsDevice graphicsDevice, List<ModelBone> bones, List<ModelMesh> meshes)
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Model.cs]: I think it is worth keeping this API public.  There is really nothing that needs to be private or hidden here.

@KonajuGames ?

Comment (Shqrdx) [inline MonoGame.Framework/Graphics/Model.cs]: In XNA there is no public constructor. I really doubt that someone needs to construct it manually from parts. These constuctors in general used only by content pipeline.

Comment (tomspilman) [inline MonoGame.Framework/Graphics/Model.cs]: I am fully aware of how they are generally used and that it isn't public in XNA.  There is zero harm in making this public anyway and allows someone to build a model from code which is reasonable in some cases.

Comment (Shqrdx) [inline MonoGame.Framework/Graphics/Model.cs]: Ok, than I open it and adds docs..

**Comment by Shqrdx** [comment]
@tomspilman Should I also opens BuildHierarchy method ?

**Comment by Shqrdx** [comment]
I guess not, because users already created model with this constructor with defined bones and this method is just confusing.

**Comment by Shqrdx** [comment]
@tomspilman Accept ?

File: MonoGame.Framework/Graphics/Model.cs (hunk 1 above)
Comment (Shqrdx) [inline MonoGame.Framework/Graphics/Model.cs]: (after test) Yes this constructor works and you can really build model from code.

**Comment by Shqrdx** [comment]
@tomspilman Recheck pls

**Comment by tomspilman** [comment]
Ok...  i'm good with this.  Merging!

**Pull request merged**